src/init.cpp

```diff
@@ -1687,26 +1687,23 @@ bool AppInitMain(const util::Ref& context, NodeContext& node, interfaces::BlockA
                 break;
             }
 
-            bool failed_rewind{false};
-            // Can't hold cs_main while calling RewindBlockIndex, so retrieve the relevant
+            bool needs_ibd{false};
+            // Can't hold cs_main while calling NeedsIBD, so retrieve the relevant
```

Comment:
This isn't true. `NeedsIBD()` grabs cs_main and holds throughout. You could change it to requires cs_main, then run this entire block under one cs_main lock.

Reply:
Done.